### PR TITLE
Add Border and String drawing

### PR DESCRIPTION
**Pull request by kkoser**
This adds functions to draw borders of windows, and to draw strings in
windows. When a window is created, a default border color of grey
(128,128,128) is used. To change this, the window_set_border_color()
function must be used, so that the border can be redrawn in the new color.

Tested by creating a window and drawing text to it.

### CODE REVIEW

File: src/window.h [hunk 1]
```diff
@@ -31,6 +31,13 @@ struct window {
 struct window *window_create(int x, int y, int width, int height);
 
 /**
+ * @brief Sets the border color of the window
+ * @details Sets the border color of the window and re-draws with the new color
+ * @param window The window whose border color to change
```
Comment (rboccabe) [inline src/window.h:37]: insert empty line in accordance with style guide

File: src/window.c [hunk 2]
```diff
@@ -59,6 +76,20 @@ void window_draw_char(struct window *w, int x, int y, char ch, struct graphics_c
...
 
+void window_draw_string(struct window *w, int x, int y, char *str, struct graphics_color fgcolor,
+                        struct graphics_color bgcolor) {
+    int pos_h = 0;
+    int pos_w = 0;
+    while (*str) {
+        window_draw_char(w, x + pos_w, y + pos_h, *str, fgcolor, bgcolor);
+        pos_w += CHARACTER_W;
+        if (*str == '\n') {
+            pos_h += CHARACTER_H;
```
Comment (rboccabe) [inline src/window.c:85]: should we reset pos_w = 0 here? right now it seems like a newline just drops you down without resetting your x position for drawing the next character.

Comment (kkoser) [inline src/window.c:85]: yes we should thanks for catching. I only tested with single line text :O

File: src/window.c [hunk 3]
```diff
@@ -59,6 +76,21 @@ void window_draw_char(struct window *w, int x, int y, char ch, struct graphics_c
     window_end_draw();
 }
 
+void window_draw_string(struct window *w, int x, int y, char *str, struct graphics_color fgcolor,
```
Comment (rboccabe) [inline src/window.c]: const char *?

File: src/window.h [hunk 4]
```diff
@@ -90,6 +97,21 @@ void window_draw_char(struct window *w, int x, int y, char ch, struct graphics_c
...
+ * the bounds of the window, it will be clipped.
+ *
+ * @param window The window to draw in
+ * @param x The x position of the top left corner of the first character
+ * @param y The y position of the top right corner of the first character
+ * @param str The text to be drawn, null terminated
+ * @param graphics_color The color to draw the characters
+ * @param graphics_color The color to draw the negative space
+ */
+void window_draw_string(struct window *w, int x, int y, char *str, struct graphics_color fgcolor,
```
Comment (rboccabe) [inline src/window.h]: const char \* as mentioned in c file above

Comment (kkoser) [inline src/window.h]: fixed

File: src/window.c (hunk 3 above)
Comment (kkoser) [inline src/window.c]: fixed

File: src/window.h (hunk 1 above)
Comment (kkoser) [inline src/window.h:37]: fixed

**Comment by zachwaterson** [comment]
Fix the newline character display and we're good!

File: src/window.c [hunk 5]
```diff
@@ -59,6 +76,23 @@ void window_draw_char(struct window *w, int x, int y, char ch, struct graphics_c
...
+                        struct graphics_color bgcolor) {
+    int pos_h = 0;
+    int pos_w = 0;
+    while (*str) {
+        if (*str == '\n') {
+            pos_h += CHARACTER_H;
+            pos_w = 0;
+            str++;
+            continue;
+        }
```
Comment (zachwaterson) [inline src/window.c:89]: Now it's fixed; LGTM! Ship!

**Pull request merged**